receiver/receivercreator/rules.go

```diff
@@ -15,10 +15,79 @@
 package receivercreator
 
 import (
+	"errors"
+	"fmt"
+	"regexp"
+
+	"github.com/antonmedv/expr"
+	"github.com/antonmedv/expr/vm"
+
 	"github.com/open-telemetry/opentelemetry-collector-contrib/extension/observer"
 )
 
-func ruleMatches(rule string, e observer.Endpoint) bool {
-	// TODO
-	return rule == "enabled"
+// rule wraps expr rule for later evaluation.
+type rule struct {
+	program *vm.Program
+}
+
+var ruleRe = regexp.MustCompile("^type.(Pod|Port)")
```

Comment:
What does this specify?

Reply:
Comment added. It's a hacky regex for seeing if the rule starts with a type check.